openvdb/openvdb/tools/Morphology.h

```diff
@@ -63,540 +59,835 @@ enum NearestNeighbors { NN_FACE = 6, NN_FACE_EDGE = 18, NN_FACE_EDGE_VERTEX = 26
-/// however (unlike dilateVoxels method below) it offers no performance
-/// advantage over the one that takes a tree. Its merely included for
-/// API compatability. The leaf nodes in the manger are updated
-/// after the dilation, which incurres a (very small) overhead.
-///
-/// @note This method is fully multi-threaded and support active tiles!
-///
-/// @param manager       Leaf node manager for the tree to be dilated.
-///                      On exit it is updated to include all the leaf
-///                      nodes of the dilated tree.
-/// @param iterations    number of iterations to apply the dilation
-/// @param nn            connectivity pattern of the dilation: either
+/// @param tree          tree or leaf manager to be eroded. The leaf
+///                      manager will be synchronized with the result.
+/// @param iterations    number of iterations to apply the erosion
+/// @param nn            connectivity pattern of the erosion: either
 ///     face-adjacent (6 nearest neighbors), face- and edge-adjacent
 ///     (18 nearest neighbors) or face-, edge- and vertex-adjacent (26
 ///     nearest neighbors).
 /// @param mode          Defined the policy for handling active tiles
 ///                      (see above for details)
-///
-/// @note The values of any voxels are unchanged.
-template<typename TreeType>
-inline void dilateActiveValues(tree::LeafManager<TreeType>& manager,
-                               int iterations = 1,
-                               NearestNeighbors nn = NN_FACE,
-                               TilePolicy mode = PRESERVE_TILES);
-
+/// @param threaded      Whether to multi-thread execution
+template<typename TreeOrLeafManagerT>
+inline void erodeActiveValues(TreeOrLeafManagerT& tree,
+    const int iterations = 1,
+    const NearestNeighbors nn = NN_FACE,
+    const TilePolicy mode = PRESERVE_TILES,
+    const bool threaded = true);
 
 /// @brief Topologically dilate all leaf-level active voxels in a tree
-/// using one of three nearest neighbor connectivity patterns.
-/// @warning This method is NOT multi-threaded and ignores active tiles!
+///   using one of three nearest neighbor connectivity patterns.
+/// @warning This method ignores active tiles
+/// @note The values of any voxels are unchanged. This method does not
+///   prune the resulting topology. It is recommended you call a variant
+///   of tools::prune afterwards to collapse dense nodes.
 ///
 /// @param tree          tree to be dilated
 /// @param iterations    number of iterations to apply the dilation
 /// @param nn            connectivity pattern of the dilation: either
 ///     face-adjacent (6 nearest neighbors), face- and edge-adjacent
 ///     (18 nearest neighbors) or face-, edge- and vertex-adjacent (26
 ///     nearest neighbors).
+/// @param threaded      Whether to multi-thread execution
+template<typename TreeOrLeafManagerT>
+inline void dilateActiveLeafValues(TreeOrLeafManagerT& tree,
```

Comment:
I remember talking about this but I can't remember the reason. I still find it rather odd that we have `dilateActiveValues(..., IGNORE_TILES)` and `dilateActiveLeafValues(...)` which are the same thing?

Reply:
I think this was to provide an easier 1:1 mapping from the old functions. I guess I could remove them?

Reply:
I'm be in favor of removing `dilateActiveLeafValues`. Perhaps a little more work to migrate, but less API to support is good?

Reply:
Removed both `dilateActiveLeafValues` and `erodeActiveLeafValues`